index.js

```diff
@@ -29,10 +29,11 @@ function connect(opts) {
   const endpoint = `${opts.region}${opts.server}`;
 
   dns.lookup(endpoint, (err, address, family) => {
-    if (!err) {
-      LOGGER.debug(`Successfully resolved DNS. address: "${address}" family: "${family}"`)
+    if (err) {
+      // crash if unable to resolve DNS to avoid possible zombie container
+      throw new Error(`Failed to resolve DNS. error: ${err}`);
```

Comment:
Since this is in an asynchronous callback, will this actually crash the container?
Also no need right as we _will_ crash the container regardless when we try to actually establish a connection and it fails?

Reply:
error should be propagated upwards 🤷 
```bash
.../tmp.6Dc0WY3n :( $ cat test.js
const dns = require('dns');

dns.lookup('google.com', (err, address, family) => { if (err) { throw new Error(err); } });

dns.lookup('isntthatafaultofthevoters.com', (err, address, family) => { if (err) { throw new Error(err); } });


setTimeout(() => {
  console.log("I survived.");
}, "1000")

.../tmp.6Dc0WY3n $
.../tmp.6Dc0WY3n $
.../tmp.6Dc0WY3n $
.../tmp.6Dc0WY3n $ node test.js
/private/var/folders/vk/0z_2f26n3d1dc81ydkd4qvkxf83x89/T/tmp.6Dc0WY3n/test.js:5
dns.lookup('isntthatafaultofthevoters.com', (err, address, family) => { if (err) { throw new Error(err); } });
                                                                                   ^

Error: Error: getaddrinfo ENOTFOUND isntthatafaultofthevoters.com
    at GetAddrInfoReqWrap.callback (/private/var/folders/vk/0z_2f26n3d1dc81ydkd4qvkxf83x89/T/tmp.6Dc0WY3n/test.js:5:90)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:60:17)
.../tmp.6Dc0WY3n :( $
.../tmp.6Dc0WY3n :( $

```


yup, we should crash, but i'm not sure what the error might be, here we're very explicit